### PR TITLE
Add sidecars to Tasks

### PR DESCRIPTION
**Pull request by sbwsg**
# Changes

Closes https://github.com/tektoncd/pipeline/issues/980

This commit adds user-defined sidecars to the TaskSpec. A sidecar is a
container intended to provide some auxiliary support to a pod's primary
containers. An example usecase is a mock API server for your app to hit
during testing.

This commit doesn't introduce any extra lifecycle management beyond that
added in #936 - sidecars will be stopped without any warning or cleanup
time. At the moment I'm thinking of leaving this as-is until the sidecars
KEP reaches general availability in Kubernetes.

Design doc: https://docs.google.com/document/d/1OiV75sIqwo77QbASiItBVFVsM5gWgxvtpuc4fLcbjBs/edit

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md)
are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes)
must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS)
and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes)
must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS),
and they must first be added
[in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Tasks can now define a list of sidecar containers to run alongside their steps
```

### CODE REVIEW

File: pkg/reconciler/taskrun/resources/cloudevent/cloud_event_controller_test.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func TestSendCloudEvents(t *testing.T) {
 				SendSuccessfully: true,
 			}
 			err := SendCloudEvents(tc.taskRun, NewFakeClient(&successfulBehaviour), logger)
-			if err == nil {
+			if err != nil {
 				t.Fatalf("Unexpected error sending cloud events: %v", err)
```
Comment (sbwsg) [inline pkg/reconciler/taskrun/resources/cloudevent/cloud_event_controller_test.go]: @afrittoli I'm a little confused by this fatal message and the `if` above it.  Is it supposed to detect `error`s or `nil`s?  Reads to me like it should be `if err != nil` but want to check with you before I merge it like this.

Comment (afrittoli) [reply pkg/reconciler/taskrun/resources/cloudevent/cloud_event_controller_test.go]: That was a silly double-error - mistake in the code, enforced by the test :S 
Thanks for catching / fixing that!

File: pkg/reconciler/taskrun/resources/pod_test.go [hunk 2]
```diff
@@ -385,6 +433,7 @@ func TestMakePod(t *testing.T) {
 			}
 		})
 	}
+
```
Comment (afrittoli) [inline pkg/reconciler/taskrun/resources/pod_test.go]: NIT:?

File: pkg/status/taskrunpod_test.go [hunk 3]
```diff
@@ -454,3 +454,84 @@ func TestUpdateStatusFromPod(t *testing.T) {
 		})
 	}
 }
+
+func TestCountSidecars(t *testing.T) {
```
Comment (afrittoli) [inline pkg/status/taskrunpod_test.go]: Nice :)

**Pull request merged**